frame/nomination-pools/src/lib.rs

```diff
@@ -904,6 +904,75 @@ impl<T: Config> BondedPool<T> {
 		Ok(points_issued)
 	}
 
+	/// Bond all of the rewards from `member` into this pool.
+	///
+	/// Similar to `try_bond_funds`, but doesn't transfer funds from an account, but instead
+	/// transfers the rewards directly from the member's reward pool to the `bonded_account`.
+	///
+	/// Returns `Ok((points_issues, bonded))`, `Err` otherwise.
+	fn try_bond_funds_from_rewards(
+		&mut self,
+		member_account: &T::AccountId,
+		member: &mut PoolMember<T>,
+		reward_pool: &mut RewardPool<T>,
+		ty: BondType,
```

Comment:
if one is bonding their rewards, then the `BondType` should be `Later`. Do you think otherwise?